tests/statistics_test.py

```diff
@@ -0,0 +1,239 @@
+        os.unsetenv(ClientProperties.STATISTICS_ENABLED.name)
+        os.unsetenv(ClientProperties.STATISTICS_PERIOD_SECONDS.name)
+        client.shutdown()
+
+    def test_statistics_period(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, self.STATS_PERIOD)
+        client = HazelcastClient(config)
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response1 = self._get_client_stats_from_server()
+        self.assertTrue(response1.success)
+        self.assertIsNotNone(response1.result)
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response2 = self._get_client_stats_from_server()
+        self.assertTrue(response2.success)
+        self.assertIsNotNone(response2.result)
+
+        self.assertNotEqual(response1, response2)
+        client.shutdown()
+
+    def test_statistics_enabled_with_negative_period(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, -1 * self.STATS_PERIOD)
+        client = HazelcastClient(config)
+
+        time.sleep(2 * self.DEFAULT_STATS_PERIOD)
+        response = self._get_client_stats_from_server()
+
+        self.assertTrue(response.success)
+        self.assertIsNotNone(response.result)
+        client.shutdown()
+
+    def test_statistics_content(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, self.STATS_PERIOD)
+
+        map_name = random_string()
+
+        near_cache_config = NearCacheConfig(map_name)
+        config.near_cache_configs[map_name] = near_cache_config
+
+        client = HazelcastClient(config)
+
+        test_map = client.get_map(map_name).blocking()
+        time.sleep(2 * self.STATS_PERIOD)
+
+        response = self._get_client_stats_from_server()
+        self.assertTrue(response.success)
+        self.assertIsNotNone(response.result)
+
+        result = response.result.decode("utf-8")
+        local_address = self._get_local_address(client)
+
+        # Check near cache ve client statistics
+        self.assertEqual(1, result.count("clientName=" + client.name))
+        self.assertEqual(1, result.count("lastStatisticsCollectionTime="))
+        self.assertEqual(1, result.count("enterprise=false"))
+        self.assertEqual(1, result.count("clientType=" + CLIENT_TYPE))
+        self.assertEqual(1, result.count("clientVersion=" + CLIENT_VERSION))
+        self.assertEqual(1, result.count("clusterConnectionTimestamp="))
+        self.assertEqual(1, result.count("clientAddress=" + local_address))
+        self.assertEqual(1, result.count("nc." + map_name + ".creationTime"))
+        self.assertEqual(1, result.count("nc." + map_name + ".evictions"))
```

Comment:
you can also validate the actual values for near cache stats, can you add them?

Reply:
I am testing it in the `test_near_cache_stats `. Should I do it there too ?